### PR TITLE
Update Action to Node 16 and handle set-output deprecation

### LINKED ISSUE #114: [BUG]
### Version
redhat-actions/push-to-registry@v2

### Describe the bug
I'm getting the following warning:

> Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### Steps to reproduce, workflow links, screenshots
See annotations section, for example here: https://github.com/redhat-actions/buildah-build/actions/runs/3560797669

### PR DESCRIPTION
**Pull request by k3rnelpan1c-dev**
### Description

This PR aims to update this GitHub Action to a recent state.
Its initial aim is to use the "new" Node16 Action runtime as the currently used Node12 runtime is phased out relatively soon (alternative to #107).
It also pins and updates the dependencies used within the Action to their latest versions (ideally this should be handled by something such as Renovate-Bot or Dependabot).
In turn this PR updates to the latest `@actions/core`, which in turn fixes the [deprecation of the `set-output` command](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).

Finally, this PR updates all Actions used within the workflows of this repositoy and adds an [`.editorconfig`](https://editorconfig.org/#pre-installed) file to help new contributors.

### Related Issue(s)

* fixes #114 
* [Actions can now run in a Node.js 16 runtime](https://github.blog/changelog/2022-05-20-actions-can-now-run-in-a-node-js-16-runtime/)
* [GitHub Actions: All Actions will begin running on Node16 instead of Node12](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)
* [GitHub Actions: Deprecating save-state and set-output commands](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

### Checklist

- [x] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [x] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [ ] This change is a patch change
- [ ] This change is a minor change
- [x] This change is a major (breaking) change*

> \* it is only major in the sense that it is a major jump from node12 to node16, and GitHub recommending bumping the major version for this

### Changes made

* update Action runtime from `node12` to `node16`
* update and pin Action dependencies
  * `@actions/core` from `^1.9.1` to `1.10.0` (addresses the `set-output` deprecation)
  * `@actions/exec` from `^1.0.4` to `1.1.1`
  * `@actions/io` from `^1.0.2` to `1.1.2`
  * `ini` from `^2.0.0` to `3.0.1`
* update and pin Action devDependencies
  * `@redhat-actions/action-io-generator` from `^1.5.0` to `1.5.0`
  * `@redhat-actions/eslint-config` from `^1.3.2` to `1.3.2`
  * `@redhat-actions/tsconfig` from `^1.1.1` to `1.2.0`
  * `@types/ini` from `^1.3.30` to `1.3.31`
  * `@types/node` from `^12` to `16.18.7`
  * `@typescript-eslint/eslint-plugin` from `^4.28.2` to `5.46.0`
  * `@typescript-eslint/parser` from `^4.28.2` to `5.46.0`
  * `@vercel/ncc` from `^0.25.1` to `0.34.0`
  * `eslint` from `^7.30.0` to `8.29.0`
  * `typescript` from `^4.3.5` to `4.9.4`
* fix build error with new TS version
* add `.editorconfig` (based on format present in repo)
* update all actions used in this repositories workflows

### CODE REVIEW

**Comment by k3rnelpan1c-dev** [comment]
sorry for not opening an issue before opening this, if it is necessary I can create one after the fact and I won't have any hard feelings if you choose to close this due to it being a requirement.

Regardless, if this is acceptable and you are interested in receiving similar PRs in the other Action repos I would be more than happy to help out.
Similarly, if you are interested in configuring Renovate-Bot for dependency updates, then I would be happy to help out. 

> Why Renovate over Dependabot?
> Simple, Renovate Bot can group and intelligently update PRs for dependencies where as Dependabot cannot and is rather spamy when it comes to opening and closing bump PRs.

**Comment by k3rnelpan1c-dev** [comment]
Regarding the ESLint, I can fix them, yet I opted to change as little code as possible within this PR as its scope was solely bumping the dependencies up to recent versions.

**Comment by k3rnelpan1c-dev** [comment]
re-based on the current master and bumped the devDependencies to their current versions

**Comment by k3rnelpan1c-dev** [comment]
it would be very unfortunate if this action would wind up dead as I see `podman` superior to docker (alone for the `oci` format support).
I will see to keep this updated and eventually fix the ESLint errors, just so it is an all green CI :smile:

**Comment by der-eismann** [comment]
We see that as well, that's why we [forked the repo](https://github.com/rebuy-de/buildah-build) to get some important changes in. We don't really plan to maintain this though, so it would be nice if this here would become more active. Quite embarrassing for an organization like RedHat if you ask me.

**Comment by k3rnelpan1c-dev** [comment]
okay, I tried to fix all the eslint errors (after the eslint updates), however, I have no idea how this ever worked or how it was intended to work:

https://github.com/redhat-actions/buildah-build/blob/4b8d36793be00cc9ee95b449fee9cd44a7a8c6e9/src/buildah.ts#L125-L131

> Error:(133, 9) ESLint: Invalid loop. Its body allows only one iteration. (no-unreachable-loop)

The rest was mostly parameter arrangement "problems" (was not sure on the code style target of the project so adhered to eslint's interpretation)

**Comment by k3rnelpan1c-dev** [comment]
> okay, I tried to fix all the eslint errors (after the eslint updates), however, I have no idea how this ever worked or how it was intended to work:
> 
> https://github.com/redhat-actions/buildah-build/blob/4b8d36793be00cc9ee95b449fee9cd44a7a8c6e9/src/buildah.ts#L125-L131
> 
> > Error:(125, 9) ESLint: Invalid loop. Its body allows only one iteration. (no-unreachable-loop)
> 
> The rest was mostly parameter arrangement "problems" (was not sure on the code style target of the project so adhered to eslint's interpretation)

well, fixed it as well as the logic :sweat_smile:

File: package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "buildah-build",
-  "version": "1.0.0",
+  "version": "3.0",
```
Comment (divyansh42) [inline package.json:3]: Probably we would not want to update this version. As there seems to be no breaking changes.

**Review by divyansh42 — CHANGES_REQUESTED** [review]
Thanks @k3rnelpan1c-dev for the detailed PR and accept my apologies for delay in the review.
Due to some other priorities, we took too much time to respond.

I have provided a minor review, please take a look.

File: package.json (hunk 1 above)
Comment (divyansh42) [inline package.json:3]:
```suggestion
  "version": "2.0",
```

**Review by divyansh42 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (k3rnelpan1c-dev) [reply package.json:3]: In regards to that, I simply followed the recommendation/best practices that GitHub provided themselves for Actions migrating from Node 12 to Node 16.

> As a result, actions authors who move from targeting Node.js 12 to Node.js 16 should release with a new major version number to communicate that there is a possible breaking change for GitHub Enterprise Server 3.3 and earlier.

https://github.blog/changelog/2022-05-20-actions-can-now-run-in-a-node-js-16-runtime/

Comment (k3rnelpan1c-dev) [reply package.json:3]: If you disagree with that notion I will revert to v2 no problem

Comment (divyansh42) [reply package.json:3]: @k3rnelpan1c-dev let us go with version `2.0.0` for now.

Comment (divyansh42) [reply package.json]: Yes, please.

**Review by divyansh42 — APPROVED** [review]
cc @jduimovich

**Comment by divyansh42** [comment]
We have created the release for the same.
Should reflect in `v2` and `v2.11`

**Comment by k3rnelpan1c-dev** [comment]
Nice! Thank you for the heads up (Renovate already started notifying me of the new version on some of my repos :rocket:)

**Pull request merged**